src/components/sidebar/dropDownMenu.tsx

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useState } from 'react';
+import Link from 'next/link';
+import styles from './dropDownMenu.module.css';
+import { MenuItemProps } from '@/types/sidebarTypes';
+import SubMenu from './subMenu';
+import { usePathname } from 'next/navigation';
+
+const DropdownMenu = ({ items }: MenuItemProps) => {
+  const [openSubmenuId, setOpenSubmenuId] = useState<string | null>(null);
+  const [selectedItemId, setSelectedItemId] = useState<string | null>(null);
+
+  const path = usePathname();
+
+  useEffect(() => {
+    const lastOpenId = localStorage.getItem('lastOpenMenuId');
+    setOpenSubmenuId(lastOpenId);
+    setSelectedItemId(lastOpenId);
+  }, []);
+
+  const toggleSubMenu = (event: React.MouseEvent, id: string) => {
+    const lastId = localStorage.getItem('lastOpenMenuId');
+    event.stopPropagation();
+    if (openSubmenuId === id) {
+      if (lastId !== id) {
+        localStorage.setItem('lastOpenMenuId', id);
+      } else {
+        setOpenSubmenuId(null);
+        localStorage.removeItem('lastOpenMenuId');
+      }
+    } else {
+      setOpenSubmenuId(id);
+      localStorage.setItem('lastOpenMenuId', id);
+    }
+  };
+
+  const selectMenu = (id: string) => {
+    if (selectedItemId !== id) {
+      setSelectedItemId(id);
+    }
+    localStorage.setItem('lastOpenMenuId', id);
+  };
```

Comment:
when converting the page to server-side rendering, we might need to significantly change that part. I'll have to look for another method.